public/app/plugins/datasource/loki/querybuilder/parsing.ts

```diff
@@ -587,7 +587,7 @@ function getBinaryModifier(
   if (node.getChild(Bool)) {
     return { isBool: true, isMatcher: false };
   } else {
-    const matcher = node.getChild(OnOrIgnoring);
+    const matcher = node.getChild(OnOrIgnoringModifier);
```

Comment:
I'm not certain this is the right node selector, I tried looking at the [Prom](https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/prometheus/querybuilder/parsing.ts#L425C1-L425C1) version, but it always returns null as well. Could be missing an edge case or something here?

Reply:
RTFM: https://grafana.com/docs/loki/latest/query/, ignoring and on are keywords, should be able to test that they are broken without this change.

Reply:
Will need unit tests for that either way before this is ready for review

Reply:
`OnOrIgnoringModifier` appears to be the correct node selector, and matches when these functions are used.
However, there are still problems converting the `on` and `ignoring` functions (which are added in the visual query with the "Binary operation with query" from code to query builder.

In the example in the docs, we see a boolean operator being used with this "nested binary operation with query"
for example: 
`max by(stream) (count_over_time({app="aggregator"}[1m])) > bool ignoring(stream) avg(count_over_time({app="aggregator"}[1m]))`

Becomes
`max by(stream) (count_over_time({app="aggregator"} [1m])) > avg(count_over_time({app="aggregator"} [1m]))`

The boolean operator and the value of the on/ignoring vector selector are stripped out when converting between code/query editors.

Reply:
Looking into why the value of the vector matching select is getting stripped out, we'll need to make a UI change to the "Binary operations with query" to include the "bool" operator (this should probably be dealt with in another PR).

Reply:
Prometheus also has the same bugs @grafana/observability-metrics

Reply:
I see lezer is still haunting you. 
Do you mind creating an issue for prometheus?

Reply:
@itsmylife yeah I can create an issue for prom, also I'll probably submit a PR for y'alls as well as the solution will likely be identical in both places

Reply:
I created https://github.com/grafana/grafana/issues/75119, but I would just ignore it until I'm wrapping up my research for loki, all of the code I'm touching so far is identical in both datasources so I might as well submit the PR for prom as well.